### DISCUSSION TITLE
Let's talk about improving profile management for the user

### DISCUSSION THREAD

**Discussion by julie777** [OrcaSlicer/OrcaSlicer#1218] [Ideas]
I am not one of the 3D printing experts who know how everything works and how to accomplish anything in the slicer. I would say that only in the last couple months have I become an intermediate. That can be a benefit, because I still remember how hard some things were when I started and I answer those same questions for others every day in the forums.

The huge number of settings, different kinds of profiles, and especially how they all relate and interact is a very complex situation. BBL changed the name of one profile type from print to process. That did not really help. They added an interface that is improved in some ways and worse in others from what they started with from Prusa.

The initial confusion for users is that they do not understand the different types of profiles, the relationships and dependencies among them, how settings interact, and what they should and should not try. A large part of why a new way of managing, and interacting with profiles would be useful is to help make this visible and understandable to the user.

Next the UI makes the usage of profiles and settings even more confusing by not presenting the information and controls in a consistent manner. This causes mental confusion. The user is both trying to understand conceptually, and trying to "make it work" at the same time.

In order to highlight some of that interface inconsistency I am just going to provide some bullet points. Later I might discuss some of the specific issues.
- Printer
    -  has a settings gear icon, but you do not change the settings there. Instead you choose what printers are available.
    - The unlabeled drop-down is how you select the profile for printer model and printer configuration
    - there is a pen/paper icon next to the chooser that provides the ability to edit settings, which are part of a separate dialog
    - if you have made changes, the only indication is a "*" in front of the name. 
    - if you want to save those changes you must open the edit dialog
    - if you want to revert those changes, you must open the edit dialog
- Filament is similar to the printer profile in usage. The chooser to select a profile and a separate dialog to edit, save or revert all or part of the changes that you have made
- Process is where it gets very confusing. 
    - It starts with a heading that is extremely dense with controls
        - There is global/objects choice
        - advanced mode
        - view all object settings, which is a non-obvious icon
        - compare profiles with an icon might not be obvious. It certainly will not be obvious that it is for all types of profiles, because it is part of the Process Section. 
    - the next line has a chooser to select the profile, followed by revert, save, and search
        - revert right next to save is dangerous, because revert does not support undo
        - save is visible right next to the chooser that shows the profile being modified
            - this is unlike the machine and filament profiles
            - this does not provide any indication that you have to save in order to be given the option to create a new profile with a new name. (I still have to overcome a bit of fear that I am destroying the original profile when I click on save.)
    - instead of a dialog for the settings, process shows them right there in the sidebar, and in the same tabbed format

There are dozens of other things I could add. I just remembered that I often hear the question "how do I get rid of a profile?". Well, it just is not obvious to most users.

Now some comments, before everyone just dismisses this post without consideration.

I want to start a discussion and see what others think and share ideas to make it better.
I am not claiming that everything currently existing is bad or should be discarded. 
I do think there are some areas that could be improved.

Let's ignore the comments that I made above for a moment and consider just something new and independent. What if there was a profile editor, not to replace how normal changes are made while slicing a model, but a way to "manage" all of the profiles. It could provide many functions that are not available, or have to be done in little pieces currently. It could be opened just like any of the other dialogs in Orca Slicer, but it could also be a separate application.

Some of the functions that I would like to see in such a tool:
- visible profile inheritance - so the user could understand how profiles work
- visible dependencies - so the user could understand how different types of profiles are related
- the ability to create, copy, delete and edit profiles, and modify their dependencies
- the ability to view differences between profiles both inheritance related and independent from each other
- the ability to move setting values from one profile to another while viewing the differences, like Beyond Compare or other graphical diff tools
- the ability create multi-use profiles - what I mean is something like the "Bambu Lab PLA Basic" which uses the same name in the slicer chooser, but has different settings based on the nozzle size.
 

That should be enough to get a discussion started.

**Comment by shayded-exe** [comment]
Great thoughts! I think this is one of the biggest issues preventing a streamlined workflow.

I want to make a comprehensive post soon with my own thoughts. Profile inheritance is an absolute must have. Also we really need the ability to set profiles per-object.

**Reply by jrbynum** [reply]
I agree. I'm almost to the point of not using it anymore. Very frustrating. Spend time making a custom profile only to start a new project and its nowhere to be found.

**Comment by julie777** [comment]
@shayded-exe 
I am less interesting in setting profiles per object, but per plate is a big plus. I often iterate through object design and keep importing the new version and printing it. Because I cannot set the profile and settings for the plate, every import requires resetting everything. I would be very happy just using different plates for different profiles.

Managing profiles is painful at the moment. In order to delete a process profile that I created for something I have to actually select it and the edit it to be able to delete it. Crazy!

I am looking forward to hearing your thoughts.

**Comment by shayded-exe** [comment]
For everyone else's reference, I've opened an issue regarding per-plate profiles #1309

**Comment by roadshow69** [comment]
It gets even more confusing if you're synching profiles. Every time I delete a profile it reappears at a  later time. I've tried turning off the sync, but then lose all of my existing custom profiles. I wish there were a way to see the synched profiles on the server side, so that I could manually remove the old profiles that I don't want to return.

I'm fairly new to this, so it might be me doing something wrong to cause the issue. It's frustrating.

**Comment by EvilGremlin** [comment]
There must be simple checkbox to detach slicing profile form machine. It's EXTREMELY annoying to redo whole custom profile just because i changed to different printer. Also user should not be prevented from deleting all system profiles. I'd really like to use some new features and not wait until Merill port it to SuSi, but these two things are giant dealbreaker to me. I'm really baffled that these obligatory things were cut down.

**Comment by outofrange** [comment]
Being able to layer multiple profiles to override specific settings of some base profile would be really neat.
Layered profiles (you could call them _traits_) don't have to set all values - empty values just won't override the values from the base profile when being merged into the final configuration.

For example, I'd have my default 0.2 layer 2 wall profile, and merge settings from a _strength_ profile (a _trait_ or something like that) to set 3 walls and denser infill, as well as a _surface finish_ profile for ironing and more consistent outer wall speeds.

Even neater: being able to do some basic calculations in overriding _trait profiles_, like `+1` on _line count_, or `*1.5` for _infill density_.

Right now, I have to maintain multiple duplicated profiles to implement this - updating all of them to change some default setting is quite cumbersome.

**Comment by dweenz** [comment]
Can I bring back a dead discussion from the dead? what about customizing the icon image per machine custom save in the newest orca?
